source/common/network/connection_impl.h

```diff
@@ -172,8 +172,8 @@ class ConnectionImpl : public virtual Connection,
 class ClientConnectionImpl : public ConnectionImpl, virtual public ClientConnection {
 public:
   ClientConnectionImpl(Event::Dispatcher& dispatcher,
-                       const Address::InstanceConstSharedPtr remote_address,
-                       const Address::InstanceConstSharedPtr source_address);
+                       const Address::InstanceConstSharedPtr& remote_address,
```

Comment:
I think "Guideline: Don’t use a const unique_ptr& as a parameter; use widget* instead." at https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/ might apply here.

Reply:
Where is there a unique_ptr? These are all shared, and we store them in the connection so they have to be passed as such?

Reply:
Sorry, it was the stuff further down from that in that same guideline section. In any case, looking further at the code, we do want to take a shared copy later, just not temporarily here, so what is there is fine actually.